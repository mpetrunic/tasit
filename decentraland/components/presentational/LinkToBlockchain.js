import React from "react";
import PropTypes from "prop-types";
import { Button, Icon } from "native-base";
import { showError, openURL, buildBlockchainUrlFromAction } from "@helpers";

const supportedNetworks = ["ropsten"];

const _openLinkOf = async action => {
  const url = await buildBlockchainUrlFromAction(action);
  try {
    await openURL(url);
  } catch (err) {
    showError(`Unable to open action link`);
  }
};

export default function LinkToBlockchain(props) {
  const { action } = props;

  //const networkName = getNetworkName();
  const networkName = "ropsten";
  const isNetworkSupported = supportedNetworks.includes(networkName);

  if (!isNetworkSupported || !action) return null;

  const openLink = () => _openLinkOf(action);
  return (
    <Button transparent onPress={openLink}>
      <Icon name="eye" />
    </Button>
  );
}

LinkToBlockchain.propTypes = {
  action: PropTypes.object,
};
