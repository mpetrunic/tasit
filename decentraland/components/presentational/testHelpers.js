import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
};
