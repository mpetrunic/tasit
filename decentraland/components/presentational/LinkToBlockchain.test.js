import React from "react";
import { shallow } from "enzyme";
import LinkToBlockchain from "./LinkToBlockchain";
import { anAction } from "@helpers/testHelpers";

describe("LinkToBlockchain", () => {
  describe("renders the component", () => {
    it("null action", async () => {
      const action = null;
      expect(shallow(<LinkToBlockchain action={action} />)).toMatchSnapshot();
    });

    it("valid action", async () => {
      const action = anAction;
      expect(shallow(<LinkToBlockchain action={action} />)).toMatchSnapshot();
    });
  });
});
