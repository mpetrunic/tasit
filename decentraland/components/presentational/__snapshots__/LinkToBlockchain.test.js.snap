// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkToBlockchain renders the component null action 1`] = `""`;

exports[`LinkToBlockchain renders the component valid action 1`] = `""`;
